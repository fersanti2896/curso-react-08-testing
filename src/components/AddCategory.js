import React, { useState } from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ({setCategories}) => { /* Recibe a las categorias */
    const [inputValue, setInputValue] = useState('');
    
    const handleInputChange = (e) => {
        setInputValue(e.target.value);
    }

    const handleSubmit = (e) => {
        e.preventDefault() /* Esto previene que se refresque la pagina */
        if(inputValue.trim().length > 2) {
            setCategories(cat => [inputValue, ...cat]); /* Comunicacion entre componentes */
            setInputValue('');
        }
        
    }

    return (
        <form onSubmit={ handleSubmit }>
            <input 
                type="text"
                value={ inputValue }
                onChange={ handleInputChange }
            />
        </form>
    );
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
