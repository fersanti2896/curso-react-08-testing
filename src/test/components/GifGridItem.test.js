import '@testing-library/jest-dom'
import React from 'react';
import { shallow } from 'enzyme'
import { GifGridItem } from '../../components/GifGridItem';

describe('Pruebas al componente GifGridItem', () => {
    let wrapper;

    const title = 'Yugioh';
    const url = 'https://localhost/algo.png';
    
    beforeEach(() => {
        wrapper = shallow(<GifGridItem title={title} url={url}/>);
    });

    test('Debe mostrar <GifGridItem /> correctamente', () => {
      expect(wrapper).toMatchSnapshot();
    });
});