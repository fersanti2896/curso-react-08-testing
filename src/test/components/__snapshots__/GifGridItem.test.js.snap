// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas al componente GifGridItem Debe mostrar <GifGridItem /> correctamente 1`] = `
<div
  className="card animate__animated animate__fadeInLeftBig"
>
  <img
    alt="Yugioh"
    src="https://localhost/algo.png"
  />
  <p>
    Yugioh
  </p>
</div>
`;
